### PR TITLE
Desktop: Disable auto updater if APPIMAGE is not set

### LINKED ISSUE #2962: Autoupdate failing on NixOS
Part of #2916. Throws `Error: Error: APPIMAGE env is not defined` during an update.

```
Trezor-Suite-20.10.1-linux.AppImage installed in /home/tsusanka//.cache/appimage-run/ed08ffd7c4d64cbd0729d34476963cdf37d68961142120f0ea52737b2b7ff43a
Checking for update
Found version 20.11.2 (url: Trezor-Suite-20.11.2-linux.AppImage)
Downloading update from Trezor-Suite-20.11.2-linux.AppImage
updater cache dir: /home/tsusanka/.cache/@trezorsuite-desktop-updater
Error: Error: APPIMAGE env is not defined
    at newError (/home/tsusanka/.cache/appimage-run/ed08ffd7c4d64cbd0729d34476963cdf37d68961142120f0ea52737b2b7ff43a/resources/app.asar/node_modules/electron-updater/node_modules/builder-util-runtime/out/index.js:212:17)
    at Object.task (/home/tsusanka/.cache/appimage-run/ed08ffd7c4d64cbd0729d34476963cdf37d68961142120f0ea52737b2b7ff43a/resources/app.asar/node_modules/electron-updater/out/AppImageUpdater.js:108:52)
    at AppImageUpdater.executeDownload (/home/tsusanka/.cache/appimage-run/ed08ffd7c4d64cbd0729d34476963cdf37d68961142120f0ea52737b2b7ff43a/resources/app.asar/node_modules/electron-updater/out/AppUpdater.js:756:25)
(node:5475) UnhandledPromiseRejectionWarning: Error: APPIMAGE env is not defined
    at newError (/home/tsusanka/.cache/appimage-run/ed08ffd7c4d64cbd0729d34476963cdf37d68961142120f0ea52737b2b7ff43a/resources/app.asar/node_modules/electron-updater/node_modules/builder-util-runtime/out/index.js:212:17)
    at Object.task (/home/tsusanka/.cache/appimage-run/ed08ffd7c4d64cbd0729d34476963cdf37d68961142120f0ea52737b2b7ff43a/resources/app.asar/node_modules/electron-updater/out/AppImageUpdater.js:108:52)
    at AppImageUpdater.executeDownload (/home/tsusanka/.cache/appimage-run/ed08ffd7c4d64cbd0729d34476963cdf37d68961142120f0ea52737b2b7ff43a/resources/app.asar/node_modules/electron-updater/out/AppUpdater.js:756:25)
(node:5475) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 2)
(node:5475) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

### PR DESCRIPTION
**Pull request by keraf**
resolves #2962

As @prusnak suggestion on #3113, this change skips the auto updater if the `APPIMAGE` environment variable is not set on Linux. It will display a warning (if that log level is enabled). It will also disable the UI for the auto-updater if the auto-updater is not loaded (not supported by platform or not fulfilling all requirements (ie APPIMAGE env var on Linux)).

This also extends the tsconfig paths from the base config for the electron source. It also add command line parameters to force enable/disable the auto-updater as well as a feature flag to control this feature. Only if the feature is enabled and supported, it will be shown on the front-end.

And there are also some changes with how modules are loaded. I simplified it a bit. Each modules gets all dependencies and it's now using an array of strings rather than object.

### CODE REVIEW

**Comment by keraf** [comment]
Switching back to "In Progress". Before this gets merged I would prefer to also visually disable the auto-updater too (settings). That would require to send some information to the front-end on whether the auto updater has been skipped or not. 

Edit: Done

File: packages/suite-desktop/src-electron/libs/logger.ts [hunk 1]
```diff
@@ -38,6 +38,14 @@ class Logger implements ILogger {
         };
 
         if (this.logLevel > 0 && this.options.writeToDisk) {
+            if (!this.options.outputPath || !this.options.outputFile) {
+                this.error(
+                    'logger',
+                    "Can't write log to file because outputFile or outputPath are invalid",
```
Comment (szymonlesisz) [inline packages/suite-desktop/src-electron/libs/logger.ts]: maybe add the `options` object to this message. to be able to track which one is invalid?

File: packages/suite-desktop/src/support/DesktopUpdater.tsx [hunk 2]
```diff
@@ -34,8 +35,17 @@ const DesktopUpdater = ({ setIsUpdateVisible }: Props) => {
     const desktopUpdate = useSelector(state => state.desktopUpdate);
 
     useEffect(() => {
-        // Don't run in dev builds (no updates to fetch, triggers errors)
-        if (isDev()) {
+        console.log('desktopUpdate', desktopUpdate);
```
Comment (szymonlesisz) [inline packages/suite-desktop/src/support/DesktopUpdater.tsx]: rm console logs

**Review by szymonlesisz — CHANGES_REQUESTED** [review]
looks good, didnt test it

File: packages/suite-desktop/src/support/DesktopUpdater.tsx (hunk 2 above)
Comment (keraf) [reply packages/suite-desktop/src/support/DesktopUpdater.tsx]: Addressed in 08966ef (squashed).

File: packages/suite-desktop/src-electron/libs/logger.ts (hunk 1 above)
Comment (keraf) [reply packages/suite-desktop/src-electron/libs/logger.ts]: Addressed in 08966ef (squashed).

**Review by slowbackspace — APPROVED** [review]

**Pull request merged**